store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -83,56 +91,71 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
+                if (!extVolInfoHandler.validateGlobalBucketID(globalS3BucketId)) {
+                    throw VolumeServiceException.INVALID_REQUEST("Volume GlobalBucketID provided is incorrect, missing or empty", VolumeServiceException.BAD_VOLUME_GLOBAL_BUCKET_ID);
+                }
+                // no validation for volume prefix as of now
+            } else if (AdminConstants.A_VOLUME_OPEN_IO.equalsIgnoreCase(storageTypeOpenIO)) {
+                // validate proxy port as positive
+                if (volInfoRequest.getVolumeExternalOpenIOInfo().getProxyPort() <= 0) {
+                    throw VolumeServiceException.INVALID_REQUEST("Proxy port can't be negative number or null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
 
-            // validate use in frequent access threshold
-            int useInFrequentAccessThreshold = volInfoRequest.getVolumeExternalInfo().getUseInFrequentAccessThreshold();
-            if (0 > useInFrequentAccessThreshold) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume UseInFrequentAccessThreshold can't be negative number", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
-            }
+                // validate account port as positive
+                if (volInfoRequest.getVolumeExternalOpenIOInfo().getAccountPort() <= 0) {
+                    throw VolumeServiceException.INVALID_REQUEST("Account port can't be negative number or null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
 
-            // validate global bucket id
-            String globalS3BucketId = volInfoRequest.getVolumeExternalInfo().getGlobalBucketConfigurationId();
-            if (false == extVolInfoHandler.validateGlobalBucketID(globalS3BucketId)) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume GlobalBucketID provided is incorrect, missing or empty", VolumeServiceException.BAD_VOLUME_GLOBAL_BUCKET_ID);
-            }
+                // validate url is empty or not
+                if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getUrl())) {
+                    throw VolumeServiceException.INVALID_REQUEST("Url can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
+
+                // validate account is empty or not
+                if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getAccount())) {
+                    throw VolumeServiceException.INVALID_REQUEST("Account can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
 
-            // no validation for volume prefix as of now
+                // validate namespace is empty or not
+                if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getNamespace())) {
```

Comment:
getNameSpace() should be called once above changes are done.

Reply:
done